design/design-doc-rbac-manager.md

```diff
@@ -0,0 +1,843 @@
+- apiGroups: [""]
+  resources: [events]
+  verbs: [get, list, watch]
+# Crossplane editors must create provider credential secrets, and may need to
+# read or otherwise interact with connection secrets.
+- apiGroups: [""]
+  resources: [secrets]
+  verbs: ["*"]
+# Crossplane editors may see which namespaces exist, but not edit them.
+- apiGroups: [""]
+  resources: [namespaces]
+  verbs: [get, list, watch]
+# Crossplane editors have full access to built in Crossplane types.
+- apiGroups:
+  - apiextensions.crossplane.io
+  - pkg.crossplane.io
+  resources: ["*"]
+  verbs: ["*"]
+# Crossplane editors have full access to all of the composite resources and
+# claims it defines.
+- apiGroups: [xr.example.org]
+  resources:
+  - examplecomposites
+  - examplecomposites/status
+  - exampleclaims
+  - exampleclaims/status
+  verbs: ["*"]
+# Crossplane viewers have full access to all resources defined by installed
+# providers.
+- apiGroups: [provider.example.org]
+  resources:
+  - examplemanageds
+  - exampleproviderconfigs
+  verbs: ["*"]
+```
+
+The `crossplane-view` role allows read-only access to all Crossplane resources.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane-view
+rules:
+# Crossplane viewers have access to view events.
+- apiGroups: [""]
+  resources: [events]
+  verbs: [get, list, watch]
+# Crossplane viewers may see which namespaces exist.
+- apiGroups: [""]
+  resources: [namespaces]
+  verbs: [get, list, watch]
+# Crossplane viewers have read-only access to built in Crossplane types.
+- apiGroups:
+  - apiextensions.crossplane.io
+  - pkg.crossplane.io
+  resources: ["*"]
+  verbs: [get, list, watch]
+# Crossplane viewers have read-only access to all of the composite resources and
+# claims it defines.
+- apiGroups: [xr.example.org]
+  resources:
+  - examplecomposites
+  - examplecomposites/status
+  - exampleclaims
+  - exampleclaims/status
+  verbs: [get, list, watch]
+# Crossplane viewers have read-only access to all resources defined by installed
+# providers.
+- apiGroups: [provider.example.org]
+  resources:
+  - examplemanageds
+  - exampleproviderconfigs
+  verbs: [get, list, watch]
+```
+
+The `crossplane-browse` role allows read-only access to composite resources and
+compositions.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: crossplane-view
+rules:
+# Crossplane browsers have access to view events.
+- apiGroups: [""]
+  resources: [events]
+  verbs: [get, list, watch]
+# Crossplane browsers have read-only access to compositions and XRDs. This
+# allows them to discover and select an appropriate composition when creating a
+# resource claim.
+- apiGroups:
+  - apiextensions.crossplane.io
+  resources: ["*"]
+  verbs: [get, list, watch]
+# Crossplane browsers have read-only access to all of the composite resources it
+# defines. This allows them to claim a statically provisioned composite
+# resource.
```

Comment:
I admittedly don't have as much knowledge of the `browse` role as the others, but I wonder if we would want a separate `ClusterRole` for each namespace such that a user can only "browse" the cluster-scoped types for which their is a claim type enabled for their namespace. Or maybe that is the intention here and I am misinterpreting? I would also be comfortable saying that this would be nice but the complexity can be deferred for the time-being.

Reply:
Yeah, let's avoid that complexity for now. I'd like to avoid generating "namespace aligned ClusterRoles" if possible. I'm hopeful that this broad 'browse' role (or even the view role) will be appropriate for 80% of folks - keep in mind that there's always the fallback option of authoring and granting your own roles.